SpatialMaxPooling.lua

```diff
@@ -33,6 +33,7 @@ function SpatialMaxPooling:updateOutput(input)
    self.ceil_mode = self.ceil_mode or false
    self.padW = self.padW or 0
    self.padH = self.padH or 0
+   self.indices = input.new()
```

Comment:
this will always malloc a new self.indices on updateOutput. You want to do: self.indices = self.indices or input.new()